tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +63,134 @@ public function test_wp_enqueue_script() {
 		$this->assertSame( '', get_echo( 'wp_print_scripts' ) );
 	}
 
+	/**
+	 * Test non standalone before with defer.
+	 *
+	 * @ticket 12009
+	 * @dataProvider data_non_standalone_before_inline_script_with_defer
+	 */
+	public function test_non_standalone_before_inline_script_with_defer( $expected, $output, $message ) {
+		$this->assertSame( $expected, $output, $message );
+	}
+
+	public function data_non_standalone_before_inline_script_with_defer() {
```

Comment:
@kt-12 missing inline doc block.

Reply:
I don't see a separate doc block for dataProvider in most of the existing test cases. e.g data_application_passwords_can_use_capability_checks_to_determine_feature_availability in /tests/phpunit/tests/auth.php. 

But, there are also functions that have added doc block e.g data_admin_bar_nodes_with_tabindex_meta in /tests/phpunit/tests/adminbar.php

Reply:
Generally speaking, data providers are meant to provide inputs into whatever functionality you are unit testing, and not just passing parameters to a PHPUnit assertion. I think that these would probably be better broken up into several individual tests instead.